suite/tests/client-interface/drx-scattergather-aarch64.cpp

```diff
@@ -1004,484 +1023,483 @@ test_ld1_scalar_plus_vector()
             { /*zt=*/0, /*pg=*/7, /*zm=*/31 },
             std::array<uint32_t, 4> { 0x00, 0x01, 0x07, 0x10 },
             std::array<uint32_t, 4> { 0, 1, 7, 10 },
-            input_data.base_addr_for_data_size(element_size_t::BYTE),
+            INPUT_DATA.base_addr_for_data_size(element_size_t::BYTE),
         },
         {
             "ld1b scalar+vector 32bit unscaled offset sxtw",
             TEST_FUNC("ld1b z1.s, p6/z, [%[base], z30.s, sxtw]"),
             { /*zt=*/1, /*pg=*/6, /*zm=*/30 },
-            std::array<uint32_t, 4> { 0x00, 0xF1, 0x18, 0x27 },
+            std::array<uint32_t, 4> { 0x00, 0xF1, 0x18, 0xF5 },
             std::array<int32_t, 4> { 0, -1, 18, 27 },
-            input_data.base_addr_for_data_size(element_size_t::BYTE),
+            INPUT_DATA.base_addr_for_data_size(element_size_t::BYTE),
         },
         {
             "ld1b scalar+vector 32bit unpacked unscaled offset uxtw",
             TEST_FUNC("ld1b z2.d, p5/z, [%[base], z29.d, uxtw]"),
             { /*zt=*/2, /*pg=*/5, /*zm=*/29 },
             std::array<uint64_t, 2> { 0x01, 0x22 },
             std::array<uint64_t, 2> { 1, 22 },
-            input_data.base_addr_for_data_size(element_size_t::BYTE),
+            INPUT_DATA.base_addr_for_data_size(element_size_t::BYTE),
         },
         {
             "ld1b scalar+vector 32bit unpacked unscaled offset sxtw",
             TEST_FUNC("ld1b z3.d, p4/z, [%[base], z28.d, sxtw]"),
             { /*zt=*/3, /*pg=*/4, /*zm=*/28 },
-            std::array<uint64_t, 2> { 0xF2, 0x29 },
+            std::array<uint64_t, 2> { 0xF2, 0xF3 },
```

Comment:
Curious why these changes in the test reference values were needed.

Reply:
Its because I changed how the input data is laid out. It used to just be a sequence of 40 values, and that was fine for testing the true scatter/gather instructions because each element can point anywhere within the input data range. For the contiguous instructions you can set the start of the contiguous range wherever you like but then you need to be able to load however many bytes of data after that base. For something like an ld4b which loads 4 vector registers that means if you have 512-bit vector registers you need `4 * (512/8) = 256` bytes so it couldn't fit in the old input data scheme.
I changed it to increase the size of the input data a lot and fill the data with a repeating 32 value pattern so we should now have enough input data to test contiguous instructions with a variety of offsets, but that change meant some of the original scatter/gather test needed updating because the input data layout was different.